patches/api/0419-Expand-PlayerDeathEvent-API.patch

```diff
@@ -0,0 +1,111 @@
++    public PlayerDeathEvent(@NotNull final Player player, @NotNull final List<ItemStack> drops, final int droppedExp, final int newExp, final int newTotalExp, final int newLevel, @Nullable final net.kyori.adventure.text.Component adventure$deathMessage, @Nullable String deathMessage, boolean doExpDrop, boolean showDeathMessage) {
+         super(player, drops, droppedExp);
+         this.newExp = newExp;
+         this.newTotalExp = newTotalExp;
+@@ -42,9 +44,12 @@ public class PlayerDeathEvent extends EntityDeathEvent {
+         this.deathMessage = deathMessage;
+         this.adventure$deathMessage = adventure$deathMessage;
+         this.doExpDrop = doExpDrop;
++        this.adventure$deathScreenMessage = adventure$deathMessage;
++        this.showDeathMessage = showDeathMessage;
+     }
+     // Paper end
+ 
++    @Deprecated // Paper
+     public PlayerDeathEvent(@NotNull final Player player, @NotNull final List<ItemStack> drops, final int droppedExp, @Nullable final String deathMessage) {
+         this(player, drops, droppedExp, 0, deathMessage);
+     }
+@@ -67,6 +72,8 @@ public class PlayerDeathEvent extends EntityDeathEvent {
+         this.newLevel = newLevel;
+         this.deathMessage = deathMessage;
+         this.adventure$deathMessage = deathMessage != null ? net.kyori.adventure.text.serializer.legacy.LegacyComponentSerializer.legacySection().deserialize(deathMessage) : null; // Paper
++        this.adventure$deathScreenMessage = this.adventure$deathMessage; // Paper
++        this.showDeathMessage = true; // Paper
+     }
+ 
+     @Deprecated // Paper
+@@ -154,6 +161,48 @@ public class PlayerDeathEvent extends EntityDeathEvent {
+     public @Nullable net.kyori.adventure.text.Component deathMessage() {
+         return this.adventure$deathMessage;
+     }
++
++    /**
++     * Set the death message that will appear on the death screen of the dying player.
++     * <p>
++     * If set to null, no message will be shown to the dying player.
++     * <p>
++     * If the message exceeds 256 characters it will be truncated.
++     *
++     * @param deathScreenMessage Message to appear on the death screen to the dying player.
++     */
++    public void deathScreenMessage(@Nullable net.kyori.adventure.text.Component deathScreenMessage) {
++        this.adventure$deathScreenMessage = deathScreenMessage;
++    }
++
++    /**
++     * Get the death message that will appear on the death screen of the dying player.
++     * By default, it is the same value as {@link #deathMessage()}.
++     *
++     * @return Message to appear on the death screen to the dying player.
++     */
++    public @Nullable net.kyori.adventure.text.Component deathScreenMessage() {
++        return this.adventure$deathScreenMessage;
++    }
++
++    /**
++     * Returns whether the death message in this event should be shown according to
++     * `showDeathMessages` game rule.
```

Comment:
Clarify that the value is **initially** set by the gamerule to show death messages. Cause plugins can change it and there's no guarantee that its the same value when a listener is called. Also clarify what `shown` means. There's a method to set the deathScreenMessage, I feel it could easily be confused. Maybe mention something about controlling the `{@link #deathMessage()}` being sent to other players. This controls both the message being sent to players and the message shown on screen to the player who died.